### PR TITLE
internal/repos: Add test case for nameWithOwner

### PR DESCRIPTION
**Pull request by indradhanush**
## Test plan

New test case only. Builds pass.

### CODE REVIEW

File: internal/repos/github_test.go [hunk 1]
```diff
@@ -337,6 +337,12 @@ func TestGithubSource_makeRepo(t *testing.T) {
 				Url:                   "https://github.com",
 				RepositoryPathPattern: "gh/{nameWithOwner}",
 			},
+		}, {
+			name: "name-with-owner",
+			schema: &schema.GitHubConnection{
+				Url:                   "https://github.com",
+				RepositoryPathPattern: "{nameWithowner}",
```
Comment (mrnugget) [inline internal/repos/github_test.go:344]: Shouldn't this be `Owner`?

```suggestion
				RepositoryPathPattern: "{nameWithOwner}",
```

File: internal/repos/testdata/golden/GithubSource_makeRepo_name-with-owner [hunk 2]
```diff
@@ -0,0 +1,74 @@
+[
+  {
+   "ID": 0,
+   "Name": "{nameWithowner}",
```
Comment (mrnugget) [inline internal/repos/testdata/golden/GithubSource_makeRepo_name-with-owner:4]: That's wrong, right? It just puts the pattern in the name. Is that what you wanted to test?

**Review by mrnugget — CHANGES_REQUESTED** [review]

File: internal/repos/testdata/golden/GithubSource_makeRepo_name-with-owner (hunk 2 above)
Comment (ryanslade) [inline internal/repos/testdata/golden/GithubSource_makeRepo_name-with-owner:4]: Weird that the template isn't been populated here.

For example, the other tests don't do this:
https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/repos/testdata/golden/GithubSource_makeRepo_path-pattern

**Comment by indradhanush** [comment]
@mrnugget, @ryanslade: Argh. I should have seen the typo myself and the error in the tests. Sorry about that. Fixed now and the generated testfile now contains the name.

**Review by ryanslade — APPROVED** [review]

**Review by mrnugget — APPROVED** [review]

**Pull request merged**